src/main/java/apoc/merge/Merge.java

```diff
@@ -0,0 +1,59 @@
+package apoc.merge;
+
+import apoc.result.*;
+import org.neo4j.graphdb.*;
+import org.neo4j.helpers.collection.Iterators;
+import org.neo4j.procedure.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Merge {
+
+    @Context
+    public GraphDatabaseService db;
+
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.merge.node(['Label'], {key:value, ...}, {key:value,...}) - merge node with dynamic labels")
+    public Stream<NodeResult> node(@Name("label") List<String> labelNames, @Name("identProps") Map<String, Object> identProps, @Name("props") Map<String, Object> props) {
```

Comment:
Let's leave that off for a feature request. In real life `ON MATCH` is used pretty rarely.